app/code/Magento/ImportExport/Model/Import/Source/FileParser/CsvParser.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\ImportExport\Model\Import\Source\FileParser;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Filesystem;
+
+class CsvParser implements ParserInterface
+{
+    /**
+     * List of CSV parsing options
+     *
+     * @var array
+     */
+    private $options = [
+        'delimiter' => ',',
+        'enclosure' => '"',
+        'escape' => '\\',
+        'null' => null
+    ];
+
+    /**
+     * File reader
+     *
+     * @var Filesystem\File\ReadInterface
+     */
+    private $file;
+
+    /**
+     * Columns of CSV file
+     *
+     * @var string[]
+     */
+    private $columns;
+
+    public function __construct(
+        Filesystem\File\ReadInterface $file,
+        $options = []
+    ) {
+        $this->file = $file;
+        $this->options = $options + $this->options;
```

Comment:
Actually, it is the other way around, as `$options` are passed in constructor via a factory and `$this->options` are default arguments for a CSV file. I can move them to Factory to remove confusion if you'd like.